internal/server/singleprocess/service_trigger.go

```diff
@@ -71,3 +76,115 @@ func (s *service) ListTriggers(
 
 	return &pb.ListTriggerResponse{Triggers: result}, nil
 }
+
+func (s *service) RunTrigger(
+	ctx context.Context,
+	req *pb.RunTriggerRequest,
+) (*pb.RunTriggerResponse, error) {
+	if err := serverptypes.ValidateRunTriggerRequest(req); err != nil {
+		return nil, err
+	}
+	log := hclog.FromContext(ctx)
+
+	runTrigger, err := s.state.TriggerGet(req.Ref)
+	if err != nil {
+		return nil, err
+	}
+
+	log = log.With("run_trigger", runTrigger.Id)
+
+	log.Debug("building run trigger job")
+
+	// Build the job(s)
+	job := &pb.Job{
+		Workspace: runTrigger.Workspace,
+		Labels:    map[string]string{"trigger/id": runTrigger.Id},
+	}
+
+	switch op := runTrigger.Operation.(type) {
+	case *pb.Trigger_Build:
+		job.Operation = &pb.Job_Build{Build: op.Build}
+	case *pb.Trigger_Push:
+		job.Operation = &pb.Job_Push{Push: op.Push}
+	case *pb.Trigger_Deploy:
+		job.Operation = &pb.Job_Deploy{Deploy: op.Deploy}
+	case *pb.Trigger_Destroy:
+		job.Operation = &pb.Job_Destroy{Destroy: op.Destroy}
+	case *pb.Trigger_Release:
+		job.Operation = &pb.Job_Release{Release: op.Release}
+	case *pb.Trigger_Up:
+		job.Operation = &pb.Job_Up{Up: op.Up}
+	case *pb.Trigger_Init:
+		job.Operation = &pb.Job_Init{Init: op.Init}
+	default:
+		return nil, status.Errorf(codes.Internal,
+			"trigger %q is configured with an unsupported operation %T", runTrigger.Id, op)
+	}
+
+	if len(req.VariableOverrides) > 0 {
+		log.Debug("variable overrides have been requested for trigger job")
+		for i, v := range req.VariableOverrides {
+			switch vType := v.Source.(type) {
+			case *pb.Variable_Cli:
```

Comment:
@krantzinator suggested maybe we could introduce a `RunTrigger` source type here too. Although the CLI type does exactly what we want here, I think.